### ISSUE TITLE
Multi-echo derivatives

### ISSUE THREAD

**Issue by effigies** [nipreps/fmriprep#2542]
* [ ] T2* map (space `boldref`?)
* [ ] All echos, with STC+HMC+SDC applied (new flag)
* [ ] Decompositions
    * [ ] _mixing.tsv separate from _desc-confounds_timeseries.tsv
    * [ ] _metrics.tsv + json (non-BEP-012)

cc @jbteves @tsalo @eurunuela for verification

**Comment by effigies** [comment]
> * [x]  T2* map (space `boldref`?)

The T2* map is the boldref as of #2547, which will be in 21.0.0. I think by BIDS, we should change the suffix to `T2starmap` in the multi-echo case.

**Comment by effigies** [comment]
> [ ] All echos, with STC+HMC+SDC applied (new flag)

Let's call this space `T2starmap`, and we will always want it if individual echos are requested. If that's the case, would we also want the optimal combination in this space, or is it possible that we'd want the OC in T1w or MNI, but not in T2starmap?

Just trying to figure out the branching logic that we're going to need and how many flags will be needed to cover the range of plausible outputs.

**Comment by tsalo** [comment]
The T2* map shouldn't be the boldref since we removed that option in #2109, right?

**Comment by effigies** [comment]
Do you want to check #2547? I'm not sure if we re-did that or not, but it's definitely what's set to be saved as a `_boldref.nii.gz` image.

**Comment by tsalo** [comment]
You're right. For posterity, here it is:

https://github.com/nipreps/fmriprep/blob/a02c1c110966fd3f357c222992aa670d234da396/fmriprep/workflows/bold/base.py#L1005-L1008

I'm not sure why that was done. In #2181, we started calculating the final boldref based on the first echo's STC+MCed data, since that echo would have the most brain coverage.

https://github.com/nipreps/fmriprep/blob/5e291b88b72d21599ea70b43f8233a4e26d321d7/fmriprep/workflows/bold/base.py#L574-L575

We stopped using the T2* map because, while it was originally thought to have better contrast (hence the old `--t2s-coreg` option), it ended up not performing very well. @emdupre does that sound right?

BTW, to respond to your other question:
>Let's call this space T2starmap, and we will always want it if individual echos are requested. If that's the case, would we also want the optimal combination in this space, or is it possible that we'd want the OC in T1w or MNI, but not in T2starmap?

I don't think there's a strong reason to have the OC data in the T2starmap/boldref space, since it's (1) easy enough to calculate when you have the echoes, echo times, and T2* map, and (2) tedana doesn't actually allow you to provide the OC data when you're running the classification workflow on an existing mixing matrix, due to (1). So it won't be much use. That said, if it doesn't add many LOC, then I don't see the harm.

**Comment by effigies** [comment]
Okay, so we should revert the `t2star_map` -> `boldref` change and these will be two separate derivatives as well.

Would people just want the raw T2* map or transformed into the `--output-space`-declared spaces, or both?

Regarding the OC, it's not really a question of LOC, and the harm would be space wasted on an unwanted derivative. So the upshot of my question and your answer is that we shouldn't use `--output-spaces T2starmap` to decide that we want the cleaned echos since people are unlikely to want all of the consequences of that. Let's assume we'll go with something like `--bold-echos`.

So let's suppose I have `--output-spaces anat --bold-echos`, I should get something like:

```
sub-01/func/sub-01_task-rest_space-anat_boldref.nii.gz
sub-01/func/sub-01_task-rest_space-anat_desc-preproc_bold.nii.gz
sub-01/func/sub-01_task-rest_space-orig_T2starmap.nii.gz
sub-01/func/sub-01_task-rest_echo-1_space-orig_desc-preproc_bold.nii.gz
sub-01/func/sub-01_task-rest_echo-2_space-orig_desc-preproc_bold.nii.gz
sub-01/func/sub-01_task-rest_echo-3_space-orig_desc-preproc_bold.nii.gz
```

Seem reasonable? (Might need to do something other than `orig`. I'll check the spec.)

**Comment by effigies** [comment]
So it looks like we're performing SDC on the optimal combination, rather than on the individual echos prior to T2*-map estimation.

To get STC+HMC+SDC echos, we can either apply the SDC to the echos after the fact (and I guess the T2* map), or we can pass in SDC'd time series to Tedana.

What is the Tedana best practice here?

**Comment by tsalo** [comment]
>Would people just want the raw T2* map or transformed into the --output-space-declared spaces, or both?

The main benefit is in running tedana, but it can also be useful for QC. I don't know if that QC needs to be in standard space, but it could help. That said, I don't think it's a big deal if the T2* map is just in boldref space. @handwerkerd do you think a standard space T2* map would be necessary?

The outputs you list seem reasonable, though having the brain mask in `orig` space would be very helpful.

>What is the Tedana best practice here?

We currently recommend using non-SDCed data in tedana, but we don't have any actual tests comparing pre-SDC vs post-SDC. I have been using the SDCed data in my own analyses, because it's easier to take STC+MC+SDCed data, run it through tedana, and then apply the necessary transforms than to grab the data before SDC. If it's easy enough to run tedana within fMRIPrep _before_ SDC, then I'd lean slightly toward that.

**Comment by effigies** [comment]
So when you get your individual echo derivatives, you would want them STC+HMC but not SDC'd? And similarly, the T2* map should not be SDC'd?

But your optimal combinations should still be SDC'd?

Would you also need the SDC warp (I think we should do this, but it will affect priorities) to be output?

**Comment by handwerkerd** [comment]
Since I'm not an fMRIprep user, I don't completely follow the specific issue at hand here. If I get the part where you're asking me a question, it's whether there might be a need to allow someone to input a T2* map that isn't generated directly from the multi-echo time series data? That is currently a viable option in tedana. I don't know if anyone has specifically evaluated if it's better or worse. My sense it that there will be situations where a separately acquired, more empirical T2* map would be useful.

**Comment by handwerkerd** [comment]
I agree that (2) is currently what we consider best practice. This is primarily because aggressive non-linear spatial warps, as typical for distortion correction, will almost definitely distort the relationship between echoes. There might be ways to adjust for this in how tedana calculates multi-echo metrics and uses them for selecting components to reject, but that's a non-trivial bit of work.

I've recently been thinking about a proposal that might address this issue. The end of the component selection process is a list of ICA components to reject. Denoising is effectively regressing those time series from the inputted data. A plausible processing pipeline would be
STC -> HMC -> tedana -> time series from rejected components
STC -> HMC+SDC  -> regress time series from rejected components to these data

This approach has not been validated or even implemented, but I was already talking about something similar to this for including respiration regressors. It might also be a way to balance the need to generate components that retain the relationship between echoes, but actually denoise the data that only had one applied spatial transform.

**Comment by handwerkerd** [comment]
I'll stay out of the prioritization Q so I might be missing something in @tsalo's comment. I do agree with @effigies that the current approach would say do both:
    EchoN -> STC -> HMC -> SDC -> Tedana just for Optimal combination
    EchoN -> STC -> HMC -> Derivative echos -> Tedana for multi-echo denoising

That said, the spatial interpolation issues with SDC might affect the quality of the optimal combination, but not enough to really care about. `Optimal combination` is the weighted average of each of the echoes and if those weights are slightly off, it's still a weighted average of the same underlying data.

**Comment by effigies** [comment]
To come back to the T2* map question:

> @effigies:
> > Would people just want the raw T2* map or transformed into the `--output-space`-declared spaces, or both?
> 
> @tsalo:
> > The main benefit is in running tedana, but it can also be useful for QC. I don't know if that QC needs to be in standard space, but it could help. That said, I don't think it's a big deal if the T2* map is just in boldref space. @handwerkerd do you think a standard space T2* map would be necessary?
>
> @handwerkerd 
> > If I get the part where you're asking me a question, it's whether there might be a need to allow someone to input a T2* map that isn't generated directly from the multi-echo time series data?

The question is "What would be useful to somebody using preprocessed ME data?" Assuming you're interested in either assessing the quality of preprocessing or doing further processing with Tedana. We can output T2* maps as directly generated by Tedana, or transform them into another space.

It seems the can be regenerated at-will from the individual echos, but would they save processing time to have available? And if they are useful for QC, in what space(s) would they be most useful to be sampled?

**Comment by tsalo** [comment]
Sorry for the delay. For the sake of _using_ ME data, I don't think there's any benefit to outputting the T2* map in anything other than the space in which tedana generated it ("scanner" or "boldref").

>It seems the can be regenerated at-will from the individual echos, but would they save processing time to have available? 

It won't generally save much time to have the T2* map saved instead of generating it again from the individual echos.

>And if they are useful for QC, in what space(s) would they be most useful to be sampled?

It could be helpful to make sure that the T2* map looks reasonable, but I don't think it's necessary. Maybe it's just something that can be added in the future if folks request it? If it was being used for QC, then I'd say that standard space would be the way to go.

**Comment by bpinsard** [comment]
I noticed that multi-echo `_boldref` derivatives are skullstriped while the single-echo aren't. I am not sure the source of that discrepancy, and it might be related to #2891 .

**Comment by effigies** [comment]
I think this is settled for now.
